packages/flutter/test/widgets/overlay_layout_builder_test.dart

```diff
@@ -140,6 +140,51 @@ void main() {
     expect(regularChildSize, childSize);
   });
 
+  testWidgets('builder callback is called when OverlayPortal rebuilds', (
+    WidgetTester tester,
+  ) async {
+    late StateSetter setState;
+    Color color = const Color(0x12345678);
+    late final OverlayEntry overlayEntry;
+    addTearDown(
+      () =>
+          overlayEntry
+            ..remove()
+            ..dispose(),
+    );
+
+    Widget builder(BuildContext _, OverlayChildLayoutInfo _) => ColoredBox(color: color);
```

Comment:
I am not sure if this code is the correct usage of builder. In a widget build method, it will be developer's job to ensure the widget rebuild if it reference a stay outside of the build method. Either turn that into changenotifier and use ChangeNotifierBuilder, or use inherited widget to do dependsOnInheritedWidget.

Reply:
I don't think that's true. In a `Builder` / `StatefulBuilder` / `LayoutBuilder` the developer should be allowed to capture free variables in the callback. The framework should rebuild the entire widget tree if the root is dirty (unless it hits a cached widget) otherwise every `builder` callback will have to be closed functions because they may use variables defined in its parent widget's `State`.

Reply:
yeah this makes sense consider that is true for other builder type widget. SGTM then